azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -334,6 +337,30 @@ public boolean shouldProxy() {
     return this.shouldProxy;
   }
 
+  private void registerCustomCredential(final Props props, final Credentials hadoopCred, final
+  String userToProxy) {
+    String credentialClassName = "unknown class";
+      try {
+        if (!props.containsKey(Constants.ConfigurationKeys.CUSTOM_CREDENTIAL_NAME)) {
```

Comment:
If you use props.getString on line 347 then I don't think you need this if statement because if Constants.ConfigurationKeys.CUSTOM_CREDENTIAL_NAME isn't defined then an UndefinedPropertyException would be thrown.

Reply:
You are right!